java/memory/memory-core/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -959,7 +1008,7 @@ private ArrowBuf allocate(int nBytes) {
         final ArrowBuf arrowBuf = BaseAllocator.this.bufferWithoutReservation(nBytes, null);
 
         listener.onAllocation(nBytes);
-        if (DEBUG) {
+        if (DEBUG && historicalLog != null) {
```

Comment:
Like the other checks above, we only need to check historicalLog, not DEBUG, right?

Reply:
Changed